lib/src/socket.dart

```diff
@@ -171,7 +171,9 @@ class PhoenixSocket {
   bool get isConnected => _ws != null && _socketState == SocketState.connected;
 
   void _connect(Completer<PhoenixSocket?> completer) async {
-    if (_ws != null) {
+    if (_ws != null &&
+        (_socketState != SocketState.connected ||
```

Comment:
@matehat 
In the case of a reconnection, we have a closed socket, then we would not attempt to reconnect.
If the socket state is unknown, closing or closed, then we are able to proceed with the reconnection of the socket.